includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -603,6 +604,29 @@ public function upload_evidence( $request ) {
 		}
 	}
 
+	/**
+	 * Create a connection token.
+	 *
+	 * @param string $request request object received.
+	 *
+	 * @return array
+	 * @throws API_Exception - If request throws.
+	 */
+	public function create_token( $request ) {
+		$response = $this->request( [], self::CONN_TOKENS_API, self::POST );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		// As an aid to mobile clients, tuck in the test_mode flag (added by the request method call above) in the response.
+		if ( is_array( $response ) ) {
+			$response['test_mode'] = WC_Payments::get_gateway()->is_in_test_mode();
+		}
```

Comment:
This logic belongs in the endpoint class instead of here IMHO.

Reply:
Sure - I was on the fence myself - will fix

Reply:
Moved in https://github.com/Automattic/woocommerce-payments/pull/1283/commits/921ddf993c523a25ce06e9364969012ccbed7a6d